pinot-core/src/main/java/org/apache/pinot/core/plan/FilterPlanNode.java

```diff
@@ -56,6 +58,10 @@
 
 
 public class FilterPlanNode implements PlanNode {
+
+  private static final Set<String> CAN_APPLY_H3_INDEX_FUNCTION_NAMES =
+      ImmutableSet.of("st_distance", "stdistance", "st_contains", "stcontains");
```

Comment:
`st_within` can also be handled if the second argument is literal

Reply:
gotcha. also updated for `st_within`.